docs/en/query_language/functions/type_conversion_functions.md

```diff
@@ -17,6 +17,81 @@
 
 Converts `value` to [Decimal](../../data_types/decimal.md) of precision `S`. The `value` can be a number or a string. The `S` (scale) parameter specifies the number of decimal places.
 
+## toDecimal*OrNull
+
+Converts an input string to the value of [Nullable(Decimal(P,S))](../../data_types/decimal.md) data type. This family of functions include:
+
+- `toDecimal32OrNull( expr, S)` — Results with Nullable(Decimal32(S)) data type.
+- `toDecimal64OrNull( expr, S)` — Results with Nullable(Decimal64(S)) data type.
+- `toDecimal128OrNull( expr, S)` — Results with Nullable(Decimal128(S)) data type.
+
+**Parameters**
+
+- `expr` — Any of ClickHouse [expressions](../syntax.md#syntax-expressions), returning a value of the [String](../../data_types/string.md) data type. For example, `'1.111'`.
+- `S` — Number of decimal places in the resulting value.
+
+**Returned value**
+
+- The value of `Nullable(Decimal(P,S))` data type with `S` decimal places, if ClickHouse could interpret input string as a number.
+- `NULL`, if ClickHouse couldn't interpret input string as a number or if the input number contains more decimal places then `S`.
+
+**Examples**
+
+```sql
+SELECT toDecimal32OrNull(toString(-1.111), 5) AS val, toTypeName(val)
+```
+```text
+┌──────val─┬─toTypeName(toDecimal32OrNull(toString(-1.111), 5))─┐
+│ -1.11100 │ Nullable(Decimal(9, 5))                            │
+└──────────┴────────────────────────────────────────────────────┘
+```
+```sql
+SELECT toDecimal32OrNull(toString(-1.111), 2) AS val, toTypeName(val)
+```
+```text
+┌──val─┬─toTypeName(toDecimal32OrNull(toString(-1.111), 2))─┐
+│ ᴺᵁᴸᴸ │ Nullable(Decimal(9, 2))                            │
+└──────┴────────────────────────────────────────────────────┘
+```
+
+
+## toDecimal*OrZero
+
+Converts an input value to the [Decimal(P,S)](../../data_types/decimal.md) data type. This family of functions include:
+
+- `toDecimal32OrZero( expr, S)` — Results with Decimal32(S) data type.
+- `toDecimal64OrZero( expr, S)` — Results with Decimal64(S) data type.
+- `toDecimal128OrZero( expr, S)` — Results with Decimal128(S) data type.
+
+**Parameters**
+
+- `expr` — Any of ClickHouse [expressions](../syntax.md#syntax-expressions), returning a value of the [String](../../data_types/string.md) data type. For example, `'1.111'`.
```

Comment:
The same. We have to accent that it's expected to have string representation of decimals here most of times. But sometimes it could be some other string. Then the functions return 0 instead of error.